File: week 2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)

{
    int letters = 0;
    int words = 1;
    int sentences = 0;
    string str = get_string("Text: ");
    //iterates over string to quantify letters
    for (int i = 0; str[i] != 0; i++)
    {
        if (isalpha(str[i]))
        {
            letters++;
        }
    }
    //iterates over string to quantify words
    for (int i = 0; str[i] != 0; i++)
    {
        if (str[i] == 32)
        {
            words++;
        }
    }
    //iterates over string to quantify sentences
    for (int i = 0; str[i] != 0; i++)
    {
        if ((str[i] == 33) || (str[i] == 63) || (str[i] == 46))
        {
            sentences++;
        }
    }
    float l = (float)letters / (float)words * 100;
    float s = (float)sentences / (float)words * 100;
    float index = 0.0588 * l - 0.296 * s - 15.8;
    //  L is the average number of letters per 100 words in the text, and S is the average number of sentences per 100 words in the text.
    int grade = round(index);
    if (grade > 15)
    {
        printf("Grade 16+\n");
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}

File: week6/readability/readability.py
from cs50 import get_string

letters = 0
words = 1
sentences = 0
str = get_string("Text: ")
len = len(str)

# iterates over string to quantify letters
for i in range(0, len):
    if str[i].isalpha():
        letters += 1

# iterates over string to quantify words
for i in range(0, len):
    if str[i] == " ":
        words += 1

# iterates over string to quantify sentences
for i in range(0, len):
    if (str[i] == ".") or (str[i] == "!") or (str[i] == "?"):
        sentences += 1

# L is the average number of letters per 100 words in the text, and S is the average number of sentences per 100 words in the text.

l = letters / words * 100.0000
s = sentences / words * 100.0000
index = 0.0588 * l - 0.296 * s - 15.8
grade = round(index)

if grade > 15:
    print("Grade 16+")
elif grade < 1:
    print("Before Grade 1")
else:
    print(f"Grade {grade}")